opentracing-mock/src/main/java/io/opentracing/mock/MockSpan.java

```diff
@@ -262,4 +276,12 @@ static long nextId() {
     static long nowMicros() {
         return System.currentTimeMillis() * 1000;
     }
+
+    private synchronized void finishedCheck(String message) {
+        if (finished) {
+            RuntimeException ex = new IllegalStateException(message);
+            errors.add(ex);
```

Comment:
not sure I completely follow why it is necessary to record the exception if it's being thrown anyway - the caller already needs to deal with it, so it doesn't need a 2nd way to be told that exception happened.

but I don't object to merging it if it's useful